shortcodes/pmpro_member.php

```diff
@@ -153,3 +153,36 @@ function pmpro_member_shortcode($atts, $content=null, $code='')
 	return $r;
 }
 add_shortcode('pmpro_member', 'pmpro_member_shortcode');
+
+/**
+ * Only allow those with the edit_users capability
+ * to use the pmpro_member shortcode in posts.
+ *
+ * @since TBD
+ * @param string $content
+ * @return string
+ */
+function pmpro_maybe_strip_member_shortcode_from_posts( $content ) {
+	if ( ! current_user_can( 'edit_users' ) ) {		
+		$content = pmpro_strip_shortcode( 'pmpro_member', $content );
+	}
+	return $content;
+}
+add_filter('content_save_pre', 'pmpro_maybe_strip_member_shortcode_from_posts' );
+add_filter('excerpt_save_pre', 'pmpro_maybe_strip_member_shortcode_from_posts' );
+
+/**
+ * Only allow those with the edit_users cabapility
+ * to use the pmpro_member shortcode in widgets.
+ *
+ * @since TBD
+ * @param array $instance
+ * @return array
+ */
+function pmpro_maybe_strip_member_shortcode_from_widgets( $instance ) {
+	if ( ! current_user_can( 'edit_users' ) ) {
+		$instance = pmpro_strip_shortcode( 'pmpro_member', $instance );		
+	}
+	return $instance;
+}
+add_filter( 'widget_update_callback', 'pmpro_maybe_strip_member_shortcode_from_widgets' );

```

Comment:
Since `pmpro_strip_shortcode()` is clever enough to take an array or a string, we could eliminate one of these functions and just move the `add_filter()` to point to the remaining one.

And then if we only have a single function that uses the `pmpro_strip_shortcode()` function, maybe the abstraction is unnecessary at that point and this entire PR could just be a single function specifically for the `[pmpro_member]` shortcode.

Neither change is necessary, but just wanted to put them out there in case you think it would be cleaner.